index.bs

```diff
@@ -3331,6 +3331,207 @@ be the same.
+more substantial differences in what they do given arguments of different types. [=Union types=], in
+contrast, are usually used in the sense that "any of the types would work in about the same way".
+
+That being said, we offer the following recommendations and examples in case of difficulties to
+determine what Web IDL language feature to use.
+
+*   When the operation must return values of different types for different argument types,
+    [=overloaded|overloading=] is almost always a better choice.
+
+    Suppose there is an operation <code class="idl">calculateWithType()</code> that accepts a single
+    argument of either {{long}}, {{DOMString}}, or <code class="idl">CalculatableInterface</code>
+    (an interface type) types, and returns a value of the same type as its argument. It would seem
+    to be clearer to write the IDL fragment using [=overloaded=] operations as
+
+    <pre highlight="webidl">
+        interface A {
+          long calculateWithType(long input);
+          DOMString calculateWithType(DOMString input);
+          CalculatableInterface calculateWithType(CalculatableInterface input);
+        };
+    </pre>
+
+    than using a [=union type=] with a [=typedef=] as
+
+    <pre highlight="webidl">
+        typedef (long or DOMString or CalculatableInterface) Calculatable;
+        interface A {
+          Calculatable calculateWithType(Calculatable input);
+        };
+    </pre>
+
+    which does not convey the fact that the return value is always of the same type as |input|.
+
+    The problem is exacerbated when one of the overloads has a return type of {{void}}, since
+    [=union types=] cannot even contain {{void}} as a [=member type=]. In that case, a return type
+    of {{any}} must be used with appropriate prose defining the return type, further decreasing
+    expressiveness.
+
+*   When the operation has significantly different semantics for different argument types or
+    lengths, [=overloaded|overloading=] is preferred.
+
+    As an example, the {{CSS/supports()}} operation of the {{CSS}} interface is defined as the
+    following IDL fragment [[CSS3-CONDITIONAL]].
+
+    <pre highlight="webidl">
+        partial interface CSS {
+          static boolean supports(CSSOMString property, CSSOMString value);
+          static boolean supports(CSSOMString conditionText);
+        };
+    </pre>
+
+    Using [=optional arguments=] one can rewrite the IDL fragment as follows:
+
+    <pre highlight="webidl">
+        partial interface CSS {
+          static boolean supports(CSSOMString propertyOrConditionText, optional CSSOMString value);
+        };
+    </pre>
+
+    It can be observed that, even though the IDL is shorter, two distinctively different concepts
+    are conflated in the first argument. This makes the second version remarkably less readable than
+    the first.
+
+    Another consideration is that many specification preprocessors allow linking to two different
```

Comment:
I think this paragraph mentions specification preprocessors when that's not the primary issue. The real issue is clarity of speccing and duplicating the overload resolution logic inside your steps. I'd just say something about how specifications are easier to write and read in this case.